configuration.py

```diff
@@ -0,0 +1,212 @@
+        if not int(str(ipv4_primary_subnet).split('/')[1]) <= 29:
+            error_list.append('Test (1.2.7): Invalid "primary_ipv4_subnet". The prefix must be <= 29.')
+            result = False
+    except ValueError:
+        ipv4_primary_subnet = ipaddress.ip_network('0.0.0.0/29')
+        error_list.append(
+            'Test (1.2.6): Invalid "primary_ipv4_subnet". It is NOT valid IP address.',
+        )
+        result = False
+
+    # IPv6 Main Subnet
+    try:
+        ipv6_subnet = ipaddress.ip_network(config['ipv6_subnet'])
+        if not int(str(ipv6_subnet).split('/')[1]) <= 48:
+            error_list.append('Test (1.2.9): Invalid "ipv6_subnet".The prefix must be <= 48.')
+            result = False
+    except ValueError:
+        ipv6_subnet = ipaddress.ip_network('::/48')
+        error_list.append('Test (1.2.8): Invalid "ipv6_subnet". It is NOT valid IP address subnet.')
+        result = False
+
+    # dns ips
+    dns_ips = config['dns_ips']
+    if dns_ips in ['', None]:
+        error_list.append('Test (1.2.10): Invalid "dns_ips".It is required.')
+        result = False
+    try:
+        for ip in dns_ips.split(','):
+            ipaddress.ip_address(ip.strip())
+    except ValueError:
+        error_list.append('Test (1.2.11): Invalid "dns_ips". It must have valid IP addresses.')
+        result = False
+
+    # IPv4 and IPv6 Link Subnet only for podnet
+    if 'podnet' in installer_type:
+        try:
+            ipv4_link_subnet = ipaddress.ip_network(config['ipv4_link_subnet'])
+            if not int(str(ipv4_link_subnet).split('/')[1]) <= 29:
+                error_list.append('Test (1.2.13): Invalid "ipv4_link_subnet". The prefix must be <= 29.')
+                result = False
+        except ValueError:
+            ipv4_link_subnet = ipaddress.ip_network('0.0.0.0/29')
+            error_list.append('Test (1.2.12): Invalid "ipv4_link_subnet". It is NOT a valid IP subnet cidr.')
+            result = False
+
+        if ipv4_link_subnet in ipv4_primary_subnet:
+            error_list.append('Test (1.2.14): "ipv4_link_subnet" must NOT be within "ipv4_primary_subnet".')
+            result = False
+
+        try:
+            ipv6_link_subnet = ipaddress.ip_network(config['ipv6_link_subnet'])
+            if not int(str(ipv6_link_subnet).split('/')[1]) <= 126:
+                error_list.append('Test (1.2.16): Invalid "ipv6_link_subnet".The prefix must be <= 126.')
+                result = False
+        except ValueError:
+            ipv6_link_subnet = ipaddress.ip_network('::/126')
+            error_list.append('Test (1.2.15): Invalid "ipv6_link_subnet".It is NOT valid IP address subnet.')
+            result = False
+
+        if ipv6_link_subnet in ipv6_subnet:
+            error_list.append('Test (1.2.17): "ipv6_link_subnet" must NOT be within "ipv6_subnet".')
+            result = False
+
+    # ceph_monitors only for podnet
```

Comment:
```suggestion
    # ceph_monitors only for an appliance that its blend includes the region flavour
```
Can you review this test please? Ceph monitors will only be in the config.json on an appliance if the blend includes region.

Also it is an optional field and can be null. The reason for this is, it is not a requirement for all regions to support Ceph